src/modules/roc_audio/sample_spec.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2021 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_audio/sample_spec.h
+//! @brief Sample specifications.
+
+#ifndef ROC_AUDIO_SAMPLE_SPEC_H_
+#define ROC_AUDIO_SAMPLE_SPEC_H_
+
+#include "roc_audio/units.h"
+#include "roc_core/time.h"
+#include "roc_packet/units.h"
+
+namespace roc {
+namespace audio {
+
+//! Default sample rate, number of samples per second.
+const size_t DefaultSampleRate = 44100;
+
+//! Default channel mask.
+const packet::channel_mask_t DefaultChannelMask = 0x3;
+
+//! Sample spec.
+class SampleSpec {
+public:
+    //! Default constructor.
+    SampleSpec();
+
+    //! Constructor with sample rate and channel mask.
+    SampleSpec(size_t sample_rate, packet::channel_mask_t channel_mask);
```

Comment:
I have a suggestion regarding constructors and getters:

* default ctor will set sample rate and channel mask to zeros (instead of 44100 and 0x3)
* non-default ctor will panic if provided sample rate or channel mask is zero
* getters will panic if returned sample rate or channel mask is zero

This way, we can earlier catch various errors with fully or partially uninitialized sample specs. We can be almost sure that if tests are not crashing, then no component is using an uninitialized field, since usually components access them early in their ctors.

If there are places where a component needs to check whether sample rate or channel mask is already initialized or not, we can add have_sample_rate() and have_channel_mask() methods for them.